packages/bugsnag_flutter/lib/src/model/event.dart

```diff
@@ -10,29 +10,71 @@ import 'stackframe.dart';
 import 'thread.dart';
 import 'user.dart';
 
+/// An Event object represents an error captured by Bugsnag and is available
+/// as a parameter on [OnErrorCallback], where individual properties can be
+/// mutated before an error report is sent to Bugsnag's API.
 class BugsnagEvent {
   String? apiKey;
+
+  /// Information extracted from the errors that caused the event can be found
+  /// in this field. The list contains at least one `BugsnagError` that
+  /// represents the thrown object.
   List<BugsnagError> errors;
+
+  /// If thread state is being captured along with the event, this field will
+  /// contain a list of [BugsnagThread] objects representing the native threads
+  /// that were active when the error was captured.
   List<BugsnagThread> threads;
+
+  /// A list of breadcrumbs leading up to the event. These values can be
+  /// accessed and amended if necessary.
   List<BugsnagBreadcrumb> breadcrumbs;
+
+  /// The context of the error. The context is a summary of what was occurring
+  /// in the application at the time of the crash, if available, such as the
+  /// visible page or screen.
   String? context;
+
+  /// The grouping hash of the event to override the default grouping on the
+  /// dashboard. All events with the same grouping hash will be grouped together
+  /// into one error. This is an advanced usage of the library and mis-using it
+  /// will cause your events not to group properly in your dashboard.
+  ///
+  /// As the name implies, this option accepts a hash of sorts.
   String? groupingHash;
   bool _unhandled;
```

Comment:
Might it be an idea to group the private fields above/below the public ones? Or alongside the fields that they are relevant to?